src/utils/featheryClient/index.ts

```diff
@@ -72,7 +74,50 @@ export const updateRegionApiUrls = (region: string) => {
   }
 };
 
+/**
+ * The number of milliseconds waited until another submitCustom call
+ */
+const SUBMIT_CUSTOM_DEBOUNCE_WINDOW = 1000;
+
 export default class FeatheryClient extends IntegrationClient {
+  /**
+   * Used to aggregate field value updates for sucessive calls to
+   * submitCustom within the debounce window
+   */
+  pendingSubmitCustomUpdates: { [key: string]: any };
+
+  /**
+   * Debounced implementation of submitCustom
+   */
+  debouncedSubmitCustom: DebouncedFunc<(override: boolean) => Promise<void>>;
+
+  /**
+   * If there is a pending invocation of submitCustom, this method calls it immediately
+   */
+  flushPendingSubmitCustomUpdates: (
+    override?: boolean
+  ) => Promise<void> | undefined;
+
+  constructor(
+    formKey = '',
+    ignoreNetworkErrors?: any,
+    draft = false,
+    bypassCDN = false
+  ) {
+    super(formKey, ignoreNetworkErrors, draft, bypassCDN);
+    this.pendingSubmitCustomUpdates = {};
+    this.debouncedSubmitCustom = debounce(
+      this._debouncedSubmitCustom.bind(this),
+      SUBMIT_CUSTOM_DEBOUNCE_WINDOW
+    );
+    this.flushPendingSubmitCustomUpdates = (override?: boolean) => {
```

Comment:
can this be statically defined as a class function instead of in the constructor dynamically

Reply:
also parameter can instead be override = true so no need for ?? true when passing it into debouncedSubmitCustom below